src/pages/mysurvey/components/SurveyCard.tsx

```diff
@@ -0,0 +1,70 @@
+import { colors } from "@toss/tds-colors";
+import { Asset, Badge, Button, ProgressBar, Text } from "@toss/tds-mobile";
+import type { ActiveSurvey, ClosedSurvey, DraftSurvey } from "./types";
+
+interface SurveyCardProps {
+	survey: DraftSurvey | ActiveSurvey | ClosedSurvey;
+	type: "draft" | "active" | "closed";
+}
+
+export const SurveyCard = ({ survey, type }: SurveyCardProps) => {
+	const badgeConfig = {
+		draft: { color: "green" as const, label: "작성중" },
+		active: { color: "blue" as const, label: "노출중" },
+		closed: { color: "elephant" as const, label: "마감" },
+	};
```

Comment:
여기 타입 추론이 꼭 필요할까요??

Reply:
Badge의 color prop이 "green" | "blue" | "elephant" 리터럴 유니온만 받을 수 있어서 객체 리터럴이 string으로 추론되면 타입 에러가 날 수 있었습니다.
그래서 전에는 color 값에 as const를 붙여 임시로 타입을 고정했었어요.
그치만 as const를 반복해서 쓰는 것보다 badgeConfig에 타입을 명시하는 게 더 깔끔해서 아래처럼 수정하겠습니다. 
좋은 지적 감사합니다~

```suggestion
const badgeConfig: Record<SurveyState, { color: BadgeColor; label: string }> = {
	const badgeConfig = {
		  draft: { color: "green", label: "작성중" },
                  active: { color: "blue", label: "노출중" },
                  closed: { color: "elephant", label: "마감" },
	};
```